src/Servers/Kestrel/Core/src/Internal/Infrastructure/ConnectionMetricsContext.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Connections;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure;
+
+internal readonly struct ConnectionMetricsContext
```

Comment:
The reason not to make this a record is that we don't want memberwise equality?

Reply:
https://github.com/dotnet/aspnetcore/issues/45859